### ISSUE TITLE
Problem with DateTime? Comparison

### ISSUE THREAD

**Issue by MayurikaD** [ValeraT1982/ObjectsComparer#10]
I was trying to add a DynamicValueComparer to compare nullable dates, but I get an exception every time I try using this below:

 var dateComparer = new DynamicValueComparer<DateTime?>
( (date1, date2, settings) => date1 != null && date2 != null && date1?.Date == date2?.Date);

Am I missing some implementation here?

**Comment by ValeraT1982** [comment]
Hi @MayurikaD. What exception do you have?

I think the issue is because you use ? but not Value property for Nullable type. Try this:

```
DateTime? dt1 = new DateTime(2019, 1, 2);
DateTime? dt2 = new DateTime(2019, 1, 2);
var dateComparer = new DynamicValueComparer<DateTime?>
                ((date1, date2, settings) => date1 != null && date2 != null && date1.Value.Date == date2.Value.Date);
            
var comparer = new Comparer<DateTime?>();
comparer.AddComparerOverride<DateTime?>(dateComparer);

var result = comparer.Compare(dt1, dt2);
```

**Comment by MayurikaD** [comment]
Hi thanks for your help. So this throws a System.ArgumentException when both (or any one) values are null.
![image](https://user-images.githubusercontent.com/31679708/60946020-ba836e00-a31f-11e9-8807-685507677143.png)

**Comment by ValeraT1982** [comment]
@MayurikaD, can you provide the callstack? In what line do you have this exception? Otherwise I'll have a look next week.

**Comment by MayurikaD** [comment]
@ValeraT1982 Hi here is the detail
![image](https://user-images.githubusercontent.com/31679708/61023827-0c8ac900-a3de-11e9-89c8-d0614f568fed.png)

**Comment by ValeraT1982** [comment]
@MayurikaD, try add date1.HasValue && date2.HasValue to the condition.

And condition should return true if both values are null.

If it doesn't work then I'll have a look next week.

**Comment by MayurikaD** [comment]
Hi, tried with the HasValue check, but it still returns the same exception.

**Comment by ValeraT1982** [comment]
It looks like an issue with nulls in DynamicValueComparer. It will be fixed in the next version. 
As a workaround you can create class which implements AbstractValueComparer<DateTime?>and use it

**Comment by ValeraT1982** [comment]
Fixed in 1.4.0
